### ISSUE TITLE
[cmd/builder] ocb fails to build with default configuration (no config value path)

### ISSUE THREAD

**Issue by jackgopack4** [open-telemetry/opentelemetry-collector#11403]
**Describe the bug**

run ocb (v0.111.0) on command line, both darwin/arm64 and linux/arm64 with local go version installed in environment

**Steps to reproduce**

download `ocb` v0.111.0 for linux/arm64 or darwin/arm64, ensure go is installed locally

run `ocb`

**What did you expect to see?**

successful collector distro built to tmp folder

**What did you see instead?**

```
2024-10-08T20:41:39.980Z        INFO    internal/command.go:125 OpenTelemetry Collector Builder {"version": "0.111.0"}
2024-10-08T20:41:39.981Z        INFO    internal/command.go:136 Using default build configuration
2024-10-08T20:41:39.982Z        INFO    builder/config.go:142   Using go        {"go-executable": "/usr/local/go/bin/go"}
2024-10-08T20:41:39.982Z        INFO    builder/main.go:101     Sources created {"path": "/tmp/otelcol-distribution2631262000"}
Error: failed to update go.mod: go subcommand failed with args '[mod tidy -compat=1.22]': exit status 1, error message: go: downloading go.opentelemetry.io/collector/confmap/provider/httpprovider v0.111.0
go: downloading go.opentelemetry.io/collector/confmap/provider/httpsprovider v0.111.0
go: downloading go.opentelemetry.io/collector/confmap/provider/yamlprovider v0.111.0
go: go.opentelemetry.io/collector/confmap/provider/httpprovider@v0.111.0: reading go.opentelemetry.io/collector/confmap/provider/httpprovider/go.mod at revision confmap/provider/httpprovider/v0.111.0: unknown revision confmap/provider/httpprovider/v0.111.0
```
**What version did you use?**

v0.111.0 binary
**What config did you use?**

default config (no config file passed)
**Environment**

OS: ubuntu 22.04, as well as macOS 14.7

**Additional context**

It appears that `httpprovider` was updated to stable version; it's likely we can update some code to handle stable versions, or update the default build config in `ocb` code.

Potentially related to [ocb go mod tidy bug v0.107.0](https://github.com/open-telemetry/opentelemetry-collector/issues/10895)

**Comment by jackgopack4** [comment]
to make things more interesting, if I download the v0.111.0 tagged source code and build ocb, it fails with the same error.
If I download main and build, this isn't an issue.

**Comment by jackgopack4** [comment]
I think https://github.com/open-telemetry/opentelemetry-collector/commit/9d6a84e0b86f79ce3b6b69f7e0d6f8d25f07021d is what fixed it. @mx-psi any chance of a v0.111.1 release?

**Comment by bogdandrutu** [comment]
See https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.111.0

**Comment by jackgopack4** [comment]
yes but v0.111.0 fails without providing a config, but most updated branch with the commit above fixes it. If we want ocb to fail without a default config that seems like a reasonable decision but it would be a change from previous behavior.
v0.112.0 will certainly fix this, just didn't know if it was worthwhile releasing in the meantime.

**Comment by mx-psi** [comment]
I have double tagged v1.17.0 also as v0.111.0 for all providers, which fixed this issue for builder v0.111.0. Thanks for reporting!
